test/test_markup.rb

```diff
@@ -760,52 +760,80 @@ def assert_markup_highlights_code(markup_class, rendered)
 
   #########################################################################
   #
-  # Metadata Blocks
+  # YAML Frontmatter
   #
   #########################################################################
 
-  test "metadata blocks" do
-    content = "a\n\n<!-- ---\ntags: [foo, bar]\n-->\n\nb"
-    output  = "<p>a</p>\n\n<p>b</p>"
-    result  = { 'tags' => '[foo, bar]' }
+  test "yaml frontmatter" do
+    content = "---\ntitle: YAML in Middle Earth\ntags: [foo, bar]\n---\nSome more content"
+    output = "Some more content\n"
+    result = {'title' => 'YAML in Middle Earth', 'tags' => ['foo', 'bar']}
 
     index = @wiki.repo.index
     index.add("Bilbo-Baggins.md", content)
-    index.commit("Add metadata")
+    index.commit("Add metadata")    
 
     page     = @wiki.page("Bilbo-Baggins")
     rendered = Gollum::Markup.new(page).render
-    assert_html_equal output, rendered
+    assert_equal output, rendered.gsub(/<(\/)?p>/,'')
```

Comment:
Had to insert the ugly `gsub`s because `kramdown` on `jruby` inserts `<p>` around the result (and not on MRI).